src/db.rs

```diff
@@ -176,9 +172,106 @@ fn get_cutoff_timestamp(days: u64) -> Option<i64> {
+        db.database("host_statistics").collection("holoport_status");
+    let val = doc! {
+        "holo_network": hs.holo_network,
+        "channel": hs.channel,
+        "holoport_model": hs.holoport_model,
+        "ssh_status": hs.ssh_status,
+        "zt_ip": hs.zt_ip,
+        "wan_ip": hs.wan_ip,
+        "holoport_id": hs.holoport_id,
+        "timestamp": hs.timestamp
+    };
+    match hp_status.insert_one(val.clone(), None).await {
+        Ok(_) => Ok(()),
+        Err(e) => Err(ApiError::Database(Debug(e))),
+    }
 }
 
+/// Ops Console DB:
+// Find registration values for host identified in the `opsconsoledb` collection `registration`
+// and determine whether the provided host pub key exists within record
+pub async fn verify_host(pub_key: String, db: &Client) -> Result<(), ApiError> {
+    let records: Collection<HostRegistration> =
+        db.database("opsconsoledb").collection("registration");
+
+    let mut host_registrations = match records.find(None, None).await {
+        Ok(cursor) => cursor,
+        Err(e) => return Err(ApiError::Database(Debug(e))),
+    };
+
+    let mut host_collection = Vec::new();
+
+    while let Some(hr) = host_registrations
+        .try_next()
+        .await
+        .map_err(Debug)
+        .map_err(ApiError::Database)?
+    {
+        host_collection.push(hr);
+    }
+
+    if let Some(found) = host_collection.iter().find_map(|hr| {
+        Some(
+            hr.registration_code
+                .iter()
+                .any(|r| r.agent_pub_keys.iter().any(|keys| keys.pub_key == pub_key)),
+        )
+    }) {
+        if found {
+            return Ok(());
+        }
+    }
+
+    Err(ApiError::MissingRecord(Error404::Info(format!(
+        "No host found with provided public key. {:?}",
+        pub_key
+    ))))
+}
+
+pub fn decode_pubkey(holoport_id: &str) -> PublicKey {
+    let decoded_pubkey = base36::decode(holoport_id).unwrap();
+    PublicKey::from_bytes(&decoded_pubkey).unwrap()
+}
+
+pub async fn add_host_stats(stats: HostStats, pool: &State<AppDbPool>) -> Result<(), ApiError> {
+    let ed25519_pubkey = decode_pubkey(&stats.holoport_id);
+
+    // Confirm host exists in registration records
+    let _ = verify_host(to_holochain_encoded_agent_key(&ed25519_pubkey), &pool.mongo)
+        .await
+        .or_else(|e| {
+            return Err(ApiError::MissingRecord(Error404::Info(format!(
+                "Provided host's holoport_id is not registered among valid hosts.  Error: {:?}",
+                e
+            ))));
+        });
+
+    // Add utc timestamp to stats payload and insert into db
+    let holoport_status = HostStats {
+        holo_network: stats.holo_network,
+        channel: stats.channel,
+        holoport_model: stats.holoport_model,
+        ssh_status: stats.ssh_status,
+        zt_ip: stats.zt_ip,
+        wan_ip: stats.wan_ip,
+        holoport_id: stats.holoport_id,
+        timestamp: Some(format!("{:?}", SystemTime::now())),
```

Comment:
From what I am seeing this formats timestamp as a string: `SystemTime { tv_sec: 1646149410, tv_nsec: 651184095 }`. I think we want it as an integer, i.e. `1646149410`

Reply:
fixed in [ee4487a](https://github.com/Holo-Host/match-service-api-rust/pull/9/commits/ee4487ae5f837366fa10fae1428b3a7fd33bc25a)